src/category.rs

```diff
@@ -0,0 +1,92 @@
+use time::Timespec;
+
+use conduit::{Request, Response};
+use pg::GenericConnection;
+use pg::rows::Row;
+
+use Model;
+use db::RequestTransaction;
+use util::{RequestUtils, CargoResult};
+
+#[derive(Clone)]
+pub struct Category {
+    pub id: i32,
+    pub category: String,
+    pub created_at: Timespec,
+    pub crates_cnt: i32,
+}
+
+#[derive(RustcEncodable, RustcDecodable)]
+pub struct EncodableCategory {
+    pub id: String,
+    pub category: String,
+    pub created_at: String,
+    pub crates_cnt: i32,
+}
+
+impl Category {
+    pub fn find_by_category(conn: &GenericConnection, name: &str)
+                            -> CargoResult<Option<Category>> {
+        let stmt = try!(conn.prepare("SELECT * FROM categories \
+                                      WHERE category = $1"));
+        let rows = try!(stmt.query(&[&name]));
+        Ok(rows.iter().next().map(|r| Model::from_row(&r)))
+    }
+
+    pub fn encodable(self) -> EncodableCategory {
+        let Category { id: _, crates_cnt, category, created_at } = self;
+        EncodableCategory {
+            id: category.clone(),
+            created_at: ::encode_time(created_at),
+            crates_cnt: crates_cnt,
+            category: category,
+        }
+    }
+}
+
+impl Model for Category {
+    fn from_row(row: &Row) -> Category {
+        Category {
+            id: row.get("id"),
+            created_at: row.get("created_at"),
+            crates_cnt: row.get("crates_cnt"),
+            category: row.get("category"),
+        }
+    }
+    fn table_name(_: Option<Category>) -> &'static str { "categories" }
+}
+
+/// Handles the `GET /categories` route.
+pub fn index(req: &mut Request) -> CargoResult<Response> {
+    let conn = try!(req.tx());
+    let (offset, limit) = try!(req.pagination(10, 100));
+    let query = req.query();
+    let sort = query.get("sort").map(|s| &s[..]).unwrap_or("alpha");
```

Comment:
`.map(|s| &s[..])` might be clearer as `.map(|s| s.as_str())` or `.map(String::as_str)`.